addons/binding/org.openhab.binding.solarview/src/main/java/org/openhab/binding/solarview/handler/solarviewHandler.java

```diff
@@ -0,0 +1,417 @@
+                if (success) {
+                    for (int i = 0; i < channels.length; i++) {
+                        updateState(new ChannelUID(getThing().getUID(), channels[i]), getChannelValue(channels[i]));
+                    }
+                }
+            } catch (Exception e) {
+                logger.error("Exception occurred during execution: {}", e.getMessage(), e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+            }
+        }, 0, refreshInterval.intValue(), TimeUnit.SECONDS);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("handleCommand({},{}) called.",channelUID.getAsString(),command.toString());
+
+        if (command instanceof RefreshType) {
+            boolean success = updateEnergyDataFromServer();
+            if (success) {
+                switch (channelUID.getId()) {
+                    case CHANNEL_KYR:
+                    case CHANNEL_KMT:
+                    case CHANNEL_KDY:
+                    case CHANNEL_KT0:
+                    case CHANNEL_PAC:
+                    case CHANNEL_UDC:
+                    case CHANNEL_IDC:
+                    case CHANNEL_UDCB:
+                    case CHANNEL_IDCB:
+                    case CHANNEL_UDCC:
+                    case CHANNEL_IDCC:
+                    case CHANNEL_UL1:
+                    case CHANNEL_IL1:
+                    case CHANNEL_TKK:
+                        updateState(channelUID, getChannelValue(channelUID.getId()));
+                        break;
+                    default:
+                        logger.error("Command received for an unknown channel: {}", channelUID.getId());
+                        break;
+                }
+            }
+        } else {
+            logger.error("Command {} is not supported for channel: {}", command, channelUID.getId());
+        }
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Collection<ConfigStatusMessage> getConfigStatus() {
+        Collection<ConfigStatusMessage> configStatus = new ArrayList<>();
+        logger.trace("getConfigStatus() called.");
+        return configStatus;
+    }
+
+    /**
+     * Ask the Solarview server for an information update of the
+     * set of information (identified by {@link queryString}) and
+     * store the results within a local storage where the information
+     * items could be retrieved by {@link getChannelValue}.
+     *
+     * @return boolean success of the information request.
+     */
+    private synchronized boolean updateEnergyDataFromServer() {
+        logger.trace("updateEnergyDataFromServer() called.");
+        final Socket clientSocket;
+        try {
+            clientSocket = new Socket(hostname, SERVER_CONNECTION_DSTPORT);
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception